### PR TITLE
feat: creation of Timer list view

### PR DESCRIPTION
**Pull request by vhu-axelor**
RM#71313

### CODE REVIEW

File: packages/apps/hr/src/features/timerSlice.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import {createAsyncThunk, createSlice} from '@reduxjs/toolkit';
+import {
+  handlerApiCall,
+  generateInifiniteScrollCases,
+} from '@axelor/aos-mobile-core';
+import {fetchTimer as _fetchTimer} from '../api/timer-api';
+
+export const fetchTimer = createAsyncThunk(
+  'timre/fetchTimer',
```
Comment (gca-axelor) [inline packages/apps/hr/src/features/timerSlice.js:27]: timer

File: packages/apps/hr/src/components/molecules/TimerDeclareButton/TimerDeclareButton.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import {Button} from '@axelor/aos-mobile-ui';
+
+const TimerDeclareButton = () => {
+  const I18n = useTranslator();
+
+  return (
+    <Button
+      iconName="calendar-check"
+      title={I18n.t('Base_Declare')}
+      onPress={() => console.log('Declare button pressed.')}
```
Comment (gca-axelor) [inline packages/apps/hr/src/components/molecules/TimerDeclareButton/TimerDeclareButton.tsx:30]: maybe needed to be transorm in js file in future if needed dispatch

File: packages/apps/hr/src/features/index.js [hunk 3]
```diff
@@ -24,3 +24,4 @@ export {expenseReducer as expense} from './expenseSlice';
 export {expenseTypeReducer as expenseType} from './expenseTypeSlice';
 export {kilometricAllowParamReducer as kilometricAllowParam} from './kilometricAllowParamSlice';
 export {projectReducer as project} from './projectSlice';
+export {timerReducer as hr_timer} from './timerSlice';
```
Comment (gca-axelor) [inline packages/apps/hr/src/features/index.js:27]: weird name but its ok

File: packages/apps/hr/src/screens/timesheet/TimerListScreen.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  const {userId} = useSelector(state => state.auth);
+
+  const fetchTimerAPI = useCallback(
+    (page = 0) => {
+      dispatch(fetchTimer({userId: userId, page: page}));
+    },
+    [dispatch, userId],
+  );
+
+  useEffect(() => {
```
Comment (gca-axelor) [inline packages/apps/hr/src/screens/timesheet/TimerListScreen.js:47]: you can put this in header action

File: packages/apps/hr/src/components/molecules/TimerDeclareButton/TimerDeclareButton.tsx (hunk 2 above)
Comment (vhu-axelor) [reply packages/apps/hr/src/components/molecules/TimerDeclareButton/TimerDeclareButton.tsx:30]: I will see this later as I may use props for onPress, so I can define the logic in `TimerListScreen.js` (like it is in `ExpenseLinesListScreen.js`)

**Review by gca-axelor — APPROVED** [review]

File: packages/apps/hr/src/features/timerSlice.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+import {createAsyncThunk, createSlice} from '@reduxjs/toolkit';
+import {
+  handlerApiCall,
+  generateInifiniteScrollCases,
+} from '@axelor/aos-mobile-core';
+import {fetchTimer as _fetchTimer} from '../api/timer-api';
+
+export const fetchTimer = createAsyncThunk(
+  'timer/fetchTimer',
```
Comment (lme-axelor) [inline packages/apps/hr/src/features/timerSlice.js:27]: Need to be the same name as reducer export

File: packages/apps/hr/src/screens/expense/ExpenseLineFormScreen.js [hunk 6]
```diff
@@ -21,17 +21,17 @@ import {FormView, fetchCompanies, useDispatch} from '@axelor/aos-mobile-core';
 import {
   createExpenseLine,
   updateExpenseLine,
-} from '../features/expenseLineSlice';
+} from '../../features/expenseLineSlice';
 import {useSelector, useTranslator} from '@axelor/aos-mobile-core';
```
Comment (lme-axelor) [inline packages/apps/hr/src/screens/expense/ExpenseLineFormScreen.js:25]: move it to top

File: packages/apps/hr/src/components/atoms/TimeCard/TimeCard.tsx [hunk 7]
```diff
@@ -30,7 +30,7 @@ interface TimeCardProps {
   task?: string;
   manufOrder?: string;
   operation?: string;
-  comment?: string;
+  comments?: string;
   date: string;
   duration: string;
```
Comment (lme-axelor) [inline packages/apps/hr/src/components/atoms/TimeCard/TimeCard.tsx:35]: wrong type

**Pull request merged**